{
   "name": "divide_ints",
   "parameter_names": [
      "a",
      "b"
   ],
   "is_method": false,
   "source_file": null,
   "lines": [
      34,
      35,
      36,
      37,
      38,
      39,
      40,
      41,
      42,
      43,
      44
   ],
   "non_code_lines": [],
   "coverage_io": {
      "0": {
         "args_before": [
            "\"10\"",
            "2"
         ],
         "args_after": {},
         "kwargs": {},
         "kwargs_after": {},
         "globals_before": {
            "error_code": 0
         },
         "globals_after": {
            "error_code": -1
         },
         "expected_result": "None",
         "expected_type": "NoneType",
         "coverage": [
            34,
            35,
            36,
            37
         ],
         "exception_type": "<class 'TypeError'>",
         "exception_message": "TypeError: Variable a='10' is not an int!",
         "constructor": "",
         "cost": 0.0,
         "testable": true
      },
      "1": {
         "args_before": [
            "3",
            "0"
         ],
         "args_after": {},
         "kwargs": {},
         "kwargs_after": {},
         "globals_before": {
            "error_code": 0
         },
         "globals_after": {
            "error_code": -3
         },
         "expected_result": "None",
         "expected_type": "NoneType",
         "coverage": [
            34,
            35,
            38,
            41,
            42,
            43
         ],
         "exception_type": "<class 'ValueError'>",
         "exception_message": "ValueError: Cannot divide by zero!",
         "constructor": "",
         "cost": 0.0,
         "testable": true
      },
      "2": {
         "args_before": [
            "6",
            "2"
         ],
         "args_after": {},
         "kwargs": {},
         "kwargs_after": {},
         "globals_before": {
            "error_code": 0
         },
         "globals_after": {
            "error_code": 0
         },
         "expected_result": "6/2=3.0",
         "expected_type": "str",
         "coverage": [
            34,
            35,
            38,
            41,
            44
         ],
         "exception_type": "",
         "exception_message": "",
         "constructor": "",
         "cost": 0.0,
         "testable": true
      },
      "3": {
         "args_before": [
            "8",
            "[]"
         ],
         "args_after": {
            "b": "[]"
         },
         "kwargs": {},
         "kwargs_after": {},
         "globals_before": {
            "error_code": 0
         },
         "globals_after": {
            "error_code": -2
         },
         "expected_result": "None",
         "expected_type": "NoneType",
         "coverage": [
            34,
            35,
            38,
            39,
            40
         ],
         "exception_type": "<class 'TypeError'>",
         "exception_message": "TypeError: Variable b=[] is not an int!",
         "constructor": "",
         "cost": 0.0,
         "testable": true
      }
   },
   "coverage_percentage": 100.0,
   "types_in_use": [
      "logging.Logger",
      "logging.Manager",
      "logging.PlaceHolder",
      "logging.RootLogger",
      "logging.StreamHandler"
   ],
   "global_vars_read_from": [
      "error_code",
      "logger"
   ],
   "global_vars_written_to": [
      "error_code"
   ],
   "unified_test_coverage": [
      34,
      35,
      36,
      37,
      38,
      39,
      40,
      41,
      42,
      43,
      44
   ],
   "needs_pytest": true,
   "exceptions_raised": [
      "<class 'TypeError'>",
      "<class 'ValueError'>"
   ],
   "callable_files": {
      "divide_ints": "C:\\Users\\James\\Documents\\CyberResources\\ddt\\tests\\procedural_division\\divide_ints.py"
   }
}